x-pack/legacy/plugins/apm/public/components/app/ServiceDetails/MetricsChart.tsx

```diff
@@ -34,7 +33,7 @@ export function MetricsChart({ start, end, chart, hoverXHandlers }: Props) {
     legendValue: formatYValue(series.overallValue)
   }));
 
-  const noHits = chart.totalHits === 0;
+  const { hoverXHandlers } = useChartsSync();
```

Comment:
What do you think about skipping the intermediary object. `hoverXHandlers` is also an odd name :D
```js
const { onHover, onMouseLeave, onSelectionEnd } = useChartsSync();
```

Consumers can still use the other approach if they want:
```js
const hoverXHandlers = useChartsSync();
```